script/build-module.js

```diff
@@ -73,31 +83,45 @@ async function parseFile (file) {
     $('h2').first().text().replace('Class: ', '')
   file.description = $('blockquote').first().text().trim()
 
+  const dirname = path.dirname(file.href)
+
   // fix HREF for relative links
   $('a').each((i, el) => {
     const href = $(el).attr('href')
     const type = hrefType(href)
+
     if (type !== 'relative' && type !== 'rooted') return
-    const dirname = path.dirname(file.href)
+
     const newHref = path.resolve(dirname, href.replace(/\.md/, ''))
+              // need to have this two lines because of
+              // `path.resolve` will return C:\\ (in Windows) if it's root (/)
+              .replace(/C:\\/g, '/')
+              .replace(/\\/g, '/')
+
     $(el).attr('href', newHref)
   })
 
   // fix SRC for relative images
   $('img').each((i, el) => {
     const baseUrl = 'https://cdn.rawgit.com/electron/electron'
-    const dirname = path.dirname(file.href)
     let src = $(el).attr('src')
     const type = hrefType(src)
+
     if (type !== 'relative' && type !== 'rooted') return
 
     // turn `../images/foo/bar.png` into `/docs/images/foo/bar.png`
+
     src = path.resolve(dirname, src)
+            // the same with line 88
```

Comment:
All the regexes here make me nervous. At the very least, let's try to minimize them by creating a re-usable function.

Reply:
Agree.